API_design.md

```diff
@@ -0,0 +1,256 @@
+ # Restful API
+
+## Header
+
+### login_header
+```
+Host: api.orange.com
+Authorization: Bearer ACCESS_TOKEN
+Content-Type: application/json
+Accept: application/json
+User-Agent: OrangeApp/1.0
+
+```
+### guest_header
+`Allow anonymous access without requiring authentication headers`
+```
+
+Content-Type: application/json
+Accept: application/json
+User-Agent: OrangeApp/1.0
+
+```
+
+### Methods：GET，POST
+`https://api.orange.com/v1/products`
+
+
+### Methods：GET，PUT，PATCH，DELETE
+`https://api.orange.com/v1/products/1`
+
+
+### Pagination  get partial data
+#### NOTE：
+`limit：Querying Data Count
+max_id：Absolute position`
+
+`https://api.orange.com/v1/products?limit=10&max_id=3`
+Retrieve 10 records starting from ID 3
+So, the frontend needs to update max_id with each request
+
+## Grid Table API Examples
+### URI
+`https://api.orange.com/v1/products`
+### Example 1（Get all data）：
```

Comment:
I think `Example` is redundant, just use "Get all data" instead.

Even better, using "Get Products".